capa/capabilities/extract_domain_and_ip.py

```diff
@@ -0,0 +1,283 @@
+from capa.features.common import FORMAT_PE, FORMAT_ELF, FORMAT_DOTNET
+from capa.features.address import Address
+from capa.features.extractors import viv, binja, pefile, elffile, dotnetfile
+from capa.render.result_document import ResultDocument
+from capa.features.extractors.base_extractor import FunctionHandle
+
+# these constants are also defined in capa.main
+# defined here to avoid a circular import
+BACKEND_VIV = "vivisect"
+BACKEND_DOTNET = "dotnet"
+BACKEND_BINJA = "binja"
+BACKEND_PEFILE = "pefile"
+
+
+def invalid_domain(string: str) -> bool:
+    """
+    supports the domain extractor functions '*_extract_domain_names'
+    
+    causes the extractor function to ignore domain-like strings
+    that have invalid top-level domains (e.g., ".exe", ".dll", etc.)
+    """
+    invalid_list = ["win", "exe", "dll", "med"]  # add more to this list
+
+    for domain in invalid_list:
+        if domain == string:
+            return False
+        
+    return True
+
+
+def default_extract_domain_names(doc: ResultDocument) -> Iterator[str]:
+    """yield web domain regex matches from list of strings"""
+    ##############
+    # ideally we probably should move the 'DOMAIN_PATTERN' out of this function's scope but
+    # then we would have to pass it as a variable to this function and that would make
+    # rendering in the main function a lot more messy
+
+    # See this Stackoverflow post that discusses the parts of this regex (http://stackoverflow.com/a/7933253/433790)
+    DOMAIN_PATTERN = r"^(?!.{256})(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+(?:[a-z]{1,63}|xn--[a-z0-9]{1,59})$"
+    ##############
+    for string in domain_ip_helpers.get_file_strings(doc):
+        # re.search only accepts 'str' on byte-like objects so
+        # we convert the type of 'string'
+        string = string.value
+        if re.search(DOMAIN_PATTERN, string):
+            if not invalid_domain(string):
+                yield string
+
+        elif is_ip_addr(string):
+            yield string
+
+
+def verbose_extract_domain_and_ip(doc: ResultDocument) -> Generator[str, None, None]:
+    """yield web domain and ip address regex matches from list of strings"""
+    ##############
+    # ideally we probably should move the 'DOMAIN_PATTERN' out of this function's scope but
+    # then we would have to pass it as a variable to this function and that would make
+    # rendering in the main function a lot more messy
+
+    # See this Stackoverflow post that discusses the parts of this regex (http://stackoverflow.com/a/7933253/433790)
+    DOMAIN_PATTERN = r"^(?!.{256})(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+(?:[a-z]{1,63}|xn--[a-z0-9]{1,59})$"
+    ##############
+    domain_counts = {}
+    ip_counts = {}
+
+    for string in domain_ip_helpers.get_file_strings(doc):
+        string = string.value
+        if re.search(DOMAIN_PATTERN, string):
+            if not invalid_domain(string):
+                try:
+                    domain_counts[string] += 1
```

Comment:
`domain_counts[string] = domain_counts.get(string, 0) + 1`

In this way you don't use a try block. Faster and better